### PR TITLE
TimeBasedLineChart chart zero y axis tick addition

### PR DESCRIPTION
**Pull request by cerinman**
This PR adds the ability to always show a zero y axis dashed line with a corresponding `0` label via two new props.

alwaysShowZeroLine - Type(boolean) - Default(false) - determines if the the zero line is drawn.
shadeBelowZero - Type(boolean) - Default(false) - determines if the area below the zero line is shaded.

@mxenabled/frontend-engineers 

Before
screen shot 2016-03-01 at 1 50 34 pm

After
screen shot 2016-03-01 at 1 50 48 pm
screen shot 2016-03-01 at 1 58 28 pm

### CODE REVIEW

File: src/components/TimeBasedLineChart.js [hunk 1]
```diff
@@ -364,6 +378,30 @@ const TimeBasedLineChart = React.createClass({
     return breakPointXValue < 0 ? 0 : breakPointXValue;
   },
 
+  _getShadedRectangleYValue () {
+    return this._getYScaleValue(0);
+  },
+
+  _getZeroLabelXValue () {
+    const maxDate = this.props.data.length ? this.props.data[this.props.data.length - 1].x : 0;
```
Comment (bsbeeks) [inline src/components/TimeBasedLineChart.js]: I'd store `this.props.data` in a `data` const.

File: src/components/TimeBasedLineChart.js [hunk 2]
```diff
@@ -364,6 +378,30 @@ const TimeBasedLineChart = React.createClass({
...
+    return this._getYScaleValue(0 - this.props.margin.top);
+  },
+
+  _getZeroLineData () {
+    const maxDate = this.props.data.length ? this.props.data[this.props.data.length - 1].x : 0;
+    const minDate = this.props.data.length ? this.props.data[0].x : 0;
+    const secondMaxDate = this.props.data.length ? this.props.data[this.props.data.length - 2].x : 0;
+    const offSet = (maxDate - secondMaxDate) / 2;
+
+    return [{ x: minDate, y: 0 }, { x: maxDate + offSet, y: 0 }];
```
Comment (bsbeeks) [inline src/components/TimeBasedLineChart.js]: Same in this method.

**Comment by bsbeeks** [comment]
This looks good. Loving how the d3 components are working out.

**Comment by cerinman** [comment]
@bsbeeks this is ready to review again!

**Comment by bsbeeks** [comment]
:shipit:

**Pull request merged**